Std/Tactic/Alias.lean

```diff
@@ -0,0 +1,165 @@
+  | reverse n => n
+
+/-- The docstring for an alias. -/
+def AliasInfo.toString : AliasInfo → String
+  | plain n => s!"**Alias** of `{n}`."
+  | forward n => s!"**Alias** of the forward direction of `{n}`."
+  | reverse n => s!"**Alias** of the reverse direction of `{n}`."
+
+
+/-- Environmant extension for registering aliases -/
+initialize aliasExt : SimpleScopedEnvExtension (Name × AliasInfo) (NameMap AliasInfo) ←
+  registerSimpleScopedEnvExtension {
+    addEntry := fun st (n, i) => st.insert n i
+    initial := {}
+  }
+
+/-- Get the alias information for a name -/
+def getAliasInfo [Monad m] [MonadEnv m] (name : Name) : m (Option AliasInfo) := do
+  return aliasExt.getState (← getEnv) |>.find? name
+
+/-- Set the alias info for a new declaration -/
+def setAliasInfo [MonadEnv m] (info : AliasInfo) (declName : Name) : m Unit :=
+  modifyEnv (aliasExt.addEntry · (declName, info))
+
+/--
+  The command `alias name := target` creates a synonym of `target` with the given name.
+
+  The command `alias ⟨fwd, rev⟩ := target` creates synonyms for the forward and reverse directions
+  of an iff theorem. Use `_` if only one direction is required.
+
+  These commands accept all modifiers and attributes that `def` and `theorem` do.
+ -/
+elab (name := alias) mods:declModifiers "alias " alias:ident " := " name:ident : command =>
+  Command.liftTermElabM do
+    let resolved ← resolveGlobalConstNoOverloadWithInfo name
+    let const ← getConstInfo resolved
+    let declMods ← elabModifiers mods
+    let (declName, _) ← mkDeclName (← getCurrNamespace) declMods alias.getId
+    let decl : Declaration := match const with
+      | Lean.ConstantInfo.thmInfo t =>
+        .thmDecl { t with
+          name := declName
+          value := mkConst resolved (t.levelParams.map mkLevelParam)
+        }
+      | Lean.ConstantInfo.defnInfo c
+      | Lean.ConstantInfo.quotInfo c
+      | Lean.ConstantInfo.inductInfo c
+      | Lean.ConstantInfo.axiomInfo c
+      | Lean.ConstantInfo.opaqueInfo c
+      | Lean.ConstantInfo.ctorInfo c
+      | Lean.ConstantInfo.recInfo c =>
+        .defnDecl { c with
+          name := declName
+          value := mkConst resolved (c.levelParams.map mkLevelParam)
+          hints := .regular 0 -- FIXME
+          safety := if declMods.isUnsafe then .unsafe else .safe
+        }
+    checkNotAlreadyDeclared declName
+    if declMods.isNoncomputable then
+      addDecl decl
+    else
+      addAndCompile decl
+    Lean.addDeclarationRanges declName {
+      range := ← getDeclarationRange (← getRef)
+      selectionRange := ← getDeclarationRange alias
+    }
+    addDocString' declName declMods.docString?
+    Term.applyAttributes declName declMods.attrs
+    let info := match ← getAliasInfo const.name with
+      | some i => i
+      | none => AliasInfo.plain const.name
+    setAliasInfo info declName
+    /- alias doesn't trigger the missing docs linter so we add a default -/
+    if (← findDocString? (← getEnv) declName).isNone then
```

Comment:
```suggestion
    if declMods.docString?.isNone then
```
Seems a bit roundabout when we just set the docstring.

Reply:
Oh nvm, this is about `inherit_doc` handling, isn't it.

Reply:
Yes it is.